extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultExtension.java

```diff
@@ -59,15 +69,54 @@ public PrivateKeyResolver privateKeyResolver() {
         return privateKeyResolver;
     }
 
+    @Provider
+    public HashicorpVaultClient hashicorpVaultClient() {
+        return hashicorpVaultClient;
+    }
+
     @Override
     public void initialize(ServiceExtensionContext context) {
-        var config = HashicorpVaultClientConfig.create(context);
-        var client = new HashicorpVaultClient(config, httpClient, typeManager.getMapper());
-
-        vault = new HashicorpVault(client, context.getMonitor());
+        scheduledExecutorService = executorInstrumentation.instrument(Executors.newSingleThreadScheduledExecutor(), NAME);
+        var monitor = context.getMonitor().withPrefix(NAME);
+        var config = HashicorpVaultConfig.create(context);
+        initHashicorpVaultClient(config, monitor);
+        vault = new HashicorpVault(hashicorpVaultClient, monitor);
         privateKeyResolver = new VaultPrivateKeyResolver(vault);
+        context.registerService(CertificateResolver.class, new HashicorpCertificateResolver(vault, monitor));
+    }
 
-        context.registerService(CertificateResolver.class, new HashicorpCertificateResolver(vault, context.getMonitor()));
+    @Override
+    public void start() {
+        if (isTokenRenewable) {
+            var tokenRenewResult = hashicorpVaultClient.renewToken();
+
+            if (tokenRenewResult.failed()) {
+                throw new EdcException("[%s] Initial token renewal failed: %s".formatted(NAME, tokenRenewResult.getFailureDetail()));
+            }
+
+            var token = tokenRenewResult.getContent();
+            hashicorpVaultClient.scheduleNextTokenRenewal(token.getTimeToLive());
+        }
+    }
+
+    @Override
+    public void shutdown() {
+        scheduledExecutorService.shutdownNow();
     }
 
+    private void initHashicorpVaultClient(HashicorpVaultConfig config, Monitor monitor) {
+        hashicorpVaultClient = new HashicorpVaultClient(config, httpClient, typeManager.getMapper(), scheduledExecutorService, monitor);
+
+        var tokenLookUpResult = hashicorpVaultClient.lookUpToken();
+
+        if (tokenLookUpResult.failed()) {
+            throw new EdcException("[%s] Initial token look up failed: %s".formatted(NAME, tokenLookUpResult.getFailureDetail()));
```

Comment:
do **not** throw an exception here as it will cause a hard crash in the runtime and prevent the connector to come up. even if the initial token lookup failed, it could be a transient error (e.g. the vault pod is not (yet) reachable in K8S) that your implementation should be able to handle.

Reply:
What do you propose?
 
- Retry X times and crash ultimatelly?
- Don't crash at all and still schedule next renew?